docs/api/wasm/extensions.md

```diff
@@ -3,13 +3,74 @@ layout: docu
 title: Extensions
 ---
 
-The default [extensions](../../extensions/overview) currently enabled in DuckDB-Wasm are Parquet and FTS. The [`httpfs` extension](../../extensions/httpfs) is a specific re-implementation that comes bundled by default. 
+DuckDB-Wasm (dynamic) extextensions loading is modeled after regular DuckDB extension loading, with a few relevant differences due to the difference in platform.
 
-JSON and Excel are build-time opt-in.
+### Format
 
-## Dynamic (= Runtime) Extension Loading
+Extensions in DuckDB are binaries to be dynamic loaded via dlopen. A cryptografical signature is appended to the binary.
+Extensions in DuckDB-Wasm are a regular Wasm file to be dynamic loaded via Emscripten's dlopen. A cryptografical signature is appended to the Wasm file as a WebAssembly custom section called `duckdb_signature`.
+This ensures the file remanins a valid WebAssembly file. Current assumption is that this custom section has to be the last one, but this can be potentially relaxed in the future.
```

Comment:
```suggestion
This ensures the file remanins a valid WebAssembly file. Currently we require this custom section to be the last one, but this can be potentially relaxed in the future.
```